lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -518,6 +715,69 @@ def _call(self, method: Callable, json_data: Dict[AnyStr, Any]):
             )
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
+    def _check_authorization(self, user: str, access: str):
+        """
+        Check whether an API call is able to access data, based on the API's
+        authorization header, the requested user, the requested access
+        policy, and the API's role.
+
+        If "user" is None, then the request is unauthenticated. READ operations
+        may be allowed, UPDATE and DELETE operations will not be allowed.
+
+        If "access" is None, READ will assume we're looking for access only to
+        public datasets.
+
+        for API_OPERATION.READ:
+
+            Any call, with or without an authenticated user token, can access
+            public data.
+
+            Any authenticated user can access their own private data.
+
+            Any authenticated ADMIN user can access any private data.
+
+        for API_OPERATION.UPDATE:
+
+            An authenticated user is required.
+
+            Any authenticated user can update their own data.
+
+            Any authenticated ADMIN user can update any data.
+
```

Comment:
Is there a reason you do not document API operations of `CREATE` and `DELETE`?

Reply:
I added the definitions on Webb's suggestion to fill out the CRUD set, but never thought to add them here since I didn't do anything beyond adding the definitions. I'll be adding a DELETE operation, at least. CREATE is currently only supported via the original `PUT`, where authentication is implicit for the user which owns the authentication token; you can always create a dataset for yourself, and there's no mechanism for an ADMIN to specify that a `PUT` should be attributed to another user. (In principle we could enable this, but creating ADMIN PUT tokens just to allow `pbench-results-move` to target a different user seems rather absurd.)

Reply:
Wow, sounds like a great description that could be added here.  =)